src/model_internal.jl

```diff
@@ -0,0 +1,83 @@
+@scoped_enum(
+    RunStatus,
+    NOT_READY = -2,
+    READY = -1,
+    SUCCESSFUL = 0,
+    RUNNING = 1,
+    FAILED = 2,
+)
+
+struct TimeSeriesCacheKey
```

Comment:
Seems out of place. This package already defines `TimeSeriesKey` which is completely different.